JDBC/Assignment 1/projectTest/TestDbMainClass.java

```diff
@@ -0,0 +1,87 @@
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import dbHelper.DbHelperClass;
+import dbmain.DbMainClass;
+import pOJOClasses.Authors;
+import pOJOClasses.Members;
+import pOJOClasses.Title;
+
+public class TestDbMainClass {
+
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+	}
+
+	@Test
+	public void testGetNoOfDeleatedRow() {
+		// there was 3 row matching the condition
+
+		DbHelperClass.initConnection();
+		DbHelperClass.setAutocommit(false);
+		int expected = 3;
+		assertEquals(expected, DbMainClass.getNoOfDeleatedRows());
+		DbHelperClass.rollback();
+		DbHelperClass.closeConnection();
```

Comment:
ACTION: Connection closing should always be a part of the API. The developer (some user of the API) might forget to close the connection in the end or if the query ends up in some exception/error then this statement won't execute consequently might lead to problems in future. Therefore connection-closing should be as important as executing query itself in the API, moreover this should be done in the "finally" block of the try-catch-finally block which will surely execute even if some exception occurred.